README.markdown

```diff
@@ -8900,6 +8900,85 @@ This API was first usable in the context of [init_by_lua*](#init_by_lua) since t
 
 This API was first enabled in the `v0.6.0` release.
 
+[Back to TOC](#nginx-api-for-lua)
+
+ngx.run_worker_thread
+----------------
+**syntax:** *ok, res1, res2, ... = ngx.run_worker_thread(threadpool, module, arg, ...)*
+
+**context:** *rewrite_by_lua&#42;, access_by_lua&#42;, content_by_lua&#42;*
+
+Wrap the [nginx worker thread](http://nginx.org/en/docs/dev/development_guide.html#threads) to execute lua function. The caller coroutine would block until the function returns.
+
+The first argument `threadpool`, specifies the nginx thread pool name.
+
+For example:
+```nginx
+thread_pool testpool threads=100;
+```
+
+The second argument `module`, specifies the lua module name to execute in the worker thread, which would returns a lua function when loaded.
```

Comment:
We need to specify what kinds of arguments we can copy over. Do you support any LuaJIT GC objects? I prefer not, since it'll be tricky to copy certain types of GC objects over to the new VM state.

Also, we need to specify what kinds of APIs are available to such in-worker-thread Lua code. Apparently most of the `ngx.` APIs won't be available in the worker thread context.

Reply:
According to your implementation you only support a limited kinds of Lua value types. We should make it explicit in the documentation.